besu/src/main/java/org/web3j/tx/PrivateTransactionManager.java

```diff
@@ -200,14 +201,7 @@ public EthGetCode getCode(
 
     public String sign(final RawPrivateTransaction rawTransaction) {
 
-        final byte[] signedMessage;
-
-        if (chainId > ChainIdLong.NONE) {
-            signedMessage =
-                    PrivateTransactionEncoder.signMessage(rawTransaction, chainId, credentials);
-        } else {
-            signedMessage = PrivateTransactionEncoder.signMessage(rawTransaction, credentials);
-        }
+        final byte[] signedMessage = txSignService.sign(rawTransaction, chainId);
```

Comment:
Hi @andrii-kl , I think this change is not ok since it gets rid of the `PrivateTransactionEncoder` to replace it with the regular `TransactionEncoder` within `TxSignServiceImpl` .

I have run some tests in Besu using `4.9.3` and indeed the signature is different.

Let me know if I am missing something, otherwise probably it would be good to open an issue for the fix.

Thanks in advance, Miguel.

Reply:
just spotted this also!

Reply:
Thanks for creating the issue @antonydenyer ! If @andrii-kl is not available I could take the implementation of the fix.